src/components/data-processing/ConfigureEmbedding/ConfigureEmbedding.jsx

```diff
@@ -217,17 +217,27 @@ const ConfigureEmbedding = (props) => {
     (state) => state.componentConfig[plots[selectedPlot].plotUuid]?.outstandingChanges,
   );
 
-  const config = useSelector(
-    (state) => state.componentConfig[plots[selectedPlot].plotUuid]?.config,
-  );
+  const plotConfigs = useSelector((state) => {
+    const plotUuids = Object.values(plots).map((plotIt) => plotIt.plotUuid);
+
+    const plotConfigsToReturn = plotUuids.reduce((acum, plotUuidIt) => {
+      // eslint-disable-next-line no-param-reassign
+      acum[plotUuidIt] = state.componentConfig[plotUuidIt]?.config;
+      return acum;
+    }, {});
+
+    return plotConfigsToReturn;
+  });
 
   const plotData = useSelector(
     (state) => state.componentConfig[plots[selectedPlot].plotUuid]?.plotData,
   );
 
+  const selectedConfig = plotConfigs[plots[selectedPlot].plotUuid];
+
   useEffect(() => {
```

Comment:
Here we do the check, `config` used to be the config for only the selected plot.

- Before we checked if `selectedPlot`'s config was loaded to decide whether to load plot A's config or not.
- Now we check if plot A's config is loaded to decide whether to load plot A's config or not.

Reply:
Line 240 in green, 230 in red)